comfy_cli/knowledge.py

```
@@ -404,6 +434,309 @@ def _index(data: dict, manifest: dict | None, *, source: str, stale: bool, path:
+    head = f"no curated knowledge for {query!r}"
+    block["nudge"] = f"{head}; covered capabilities: {', '.join(sorted(bundle.capabilities))}"
+    if _block_bytes(block) > MAX_BLOCK_BYTES:
+        block["nudge"] = head
+    if _block_bytes(block) > MAX_BLOCK_BYTES:
+        del block["nudge"]
+
+
+def _log_query(command: str, query: str, block: dict, bundle: Bundle) -> None:
+    """Feed the curation miss log. Consent-gated and best-effort, like every other event."""
+    try:
+        from comfy_cli import tracking
```

Pylint (4.0.6)

[convention] Import outside toplevel (comfy_cli.tracking)

(C0415)